machineid/__init__.py

```diff
@@ -74,7 +74,11 @@ def id() -> str:
     id = __read__('/var/lib/dbus/machine-id')
     if not id:
       id = __read__('/etc/machine-id')
-
+    if not id:
+      id = __exec__('head -1 /proc/self/cgroup|cut -d/ -f3')
+    if not id:
+      id = __exec__("grep 'systemd' /proc/self/mountinfo|cut -d/ -f3")
```

Comment:
Got more info [here](https://stackoverflow.com/questions/20995351/how-can-i-get-docker-linux-container-information-from-within-the-container-itsel). I think this is fine as long as you add a similar `docker` check as above.

Reply:
Also more discussion with the above workarounds is at https://github.com/opencontainers/runtime-spec/issues/1105.

Reply:
Quick question, if `__read__('/proc/self/mountinfo')` returns `None` (because the file does not exist).
Will `if 'docker' in mountinfo:` return an **exception**?

Reply:
You’re right. I was only supplying pseudo code. I guess we should also check `if mountinfo and 'docker' in mountinfo:`. We shouldn’t raise an exception.

Reply:
You also need to fix the indentation (you're using 4 spaces but the rest of the file uses 2 spaces).

Please make sure the code runs.

Reply:
> You’re right. I was only supplying pseudo code. I guess we should also check `if mountinfo and 'docker' in mountinfo:`. We shouldn’t raise an exception.

There is no additional exception.

Currently on a Docker container the `if 'docker' in cgroup:` does not find `docker`, the ending `Exception` works as intended. Same test with `mountinfo`.

When the full code does not find an ID raises the following:

```
Python 3.11.0 (main, Nov 15 2022, 19:58:01) [GCC 10.2.1 20210110] on linux                                                    
Type "help", "copyright", "credits" or "license" for more information.                                                        
>>> import machineid                                                                                                          
>>> print(machineid.id())                                                                                                     
Traceback (most recent call last):                                                                                            
  File "<stdin>", line 1, in <module>                                                                                         
  File "/usr/local/lib/python3.11/site-packages/machineid/__init__.py", line 90, in id                                        
    raise Exception('failed to obtain id on platform {}'.format(platform))                                                    
Exception: failed to obtain id on platform linux
```